letsencrypt/cli.py

```diff
@@ -159,7 +163,34 @@ def _init_le_client(args, config, authenticator, installer):
     return client.Client(config, acc, authenticator, installer, acme=acme)
 
 
-def run(args, config, plugins):
+def _find_duplicative_certs(domains, config, renew_config):
+    """Find existing certs that duplicate the request."""
+
+    identical_names_cert, subset_names_cert = None, None
+
+    configs_dir = renew_config.renewal_configs_dir
+    for renewal_file in os.listdir(configs_dir):
+        full_path = os.path.join(configs_dir, renewal_file)
+        rc_config = configobj.ConfigObj(renew_config.renewer_config_file)
+        rc_config.merge(configobj.ConfigObj(full_path))
+        rc_config.filename = full_path
+        cli_config = configuration.RenewerConfiguration(config)
+
+        candidate_lineage = storage.RenewableCert(rc_config, None, cli_config)
+        # TODO: Handle these differently depending on whether they are
+        #       expired or still valid?
+        candidate_names = set(candidate_lineage.names())
+        if candidate_names == set(domains):
+            identical_names_cert = (renewal_file, candidate_lineage)
+        elif candidate_names.issubset(set(domains)):
+            subset_names_cert = (renewal_file, candidate_lineage,
+                                 candidate_lineage.names())
+
+    return identical_names_cert, subset_names_cert
```

Comment:
Something is wrong here... This function seems to return data only for the last `renewal_file` in `os.listdir(configs_dir)`, because you don't accumulate results in `identical_names_cert` and `subset_names_cert` for each loop iteration, but overwrite it instead.